hicommonwealth/commonwealth
# React on Content + Weighted Reactions

## Link to Issue
Depends on: #6324 
Depends on: #6313 
Closes: #5894
Closes: #5883

## Description of Changes
- Adds calculated voting weight to reactions
- Adds reaction weight sum to threads and comments + hooks to update them

## Test Plan
NOTE: Proper testing requires the blockchain namespace to be setup. It's not feasible to test locally, so we can use an override env var instead.
- Run migration
- Set env `REACTION_WEIGHT_OVERRIDE=300`
- Create a thread reaction
  - Check DB to confirm `reaction.calculated_voting_weight` is `300`
  - Check DB to confirm `thread.reaction_weights_sum` is `300`
  - Delete the thread reaction
    - Check DB to confirm `thread.reaction_weights_sum` is `0`
- Create a comment reaction
  - Check DB to confirm `reaction.calculated_voting_weight` is `300`
  - Check DB to confirm `comment.reaction_weights_sum` is `300`
  - Delete the thread reaction
    - Check DB to confirm `comment.reaction_weights_sum` is `0`

## Deployment Plan
N/A

## Other Considerations
N/A
